Feature/change value

shinchanat opened the pull request:
Added change_value function.it takes three param dict obj,key,value.This function replaces a particular key's value and returns back the modified dictionary. It works for both nested and non nested dictionary.

shinchanat committed:
Delete README.md

shinchanat committed:
Add files via upload

shinchanat committed:
Update README.md

shinchanat committed:
Update README.md

shinchanat committed:
Update README.md

shinchanat committed:
Add files via upload

shinchanat committed:
Update __init__.py

shinchanat committed:
Update __init__.py

Imported change_value function from change_value module.

shinchanat committed:
Update change_value.py

shinchanat committed:
Update README.md

shinchanat committed:
Update README.md

shinchanat committed:
Update change_value.py

shinchanat committed:
Update change_value.py

shinchanat committed:
Update change_value.py

shinchanat committed:
Update change_value.py

abhisaini880 commented on nestdict/change_value.py:
```diff
@@ -0,0 +1,62 @@
+    
+    if len(keys)>1:
+        for element in keys:
+            obj = obj[element]
```
return None or original obj if element not found in obj.
current code will throw key error.

Harish-developments replied:
Solved key error problem.

Harish-developments replied:
Solved key error problem. It will return None if element not found.

shinchanat replied:
It returns the original obj if element not found.

abhisaini880 reviewed (changes requested):
Please make the changes to handle keyerror on line 51 of change value function

shinchanat committed:
Delete change_value.py

shinchanat committed:
Add files via upload

Handled key error.

Harish-developments commented:
Handled key error,  it returns None if element not found in obj.

On Fri, Nov 12, 2021, 12:03 AM Abhishek Saini ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Please make the changes to handle keyerror on line 51 of change value
> function
> ------------------------------
>
> In nestdict/change_value.py
> <https://github.com/abhisaini880/nestdict/pull/2#discussion_r747728433>:
>
> > +       original : {"cars_owned":{"sedan":1,"suv":3},}
> +
> +       {"cars_owned":{"sedan":1,"suv":100},}
> +
> +
> +
> +      for more understanding read README.md
> +      """
> +
> +    changed_obj = obj
> +
> +    keys = list(*args)
> +
> +    if len(keys)>1:
> +        for element in keys:
> +            obj = obj[element]
>
> return None or original obj if element not found in obj.
> current code will throw key error.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/abhisaini880/nestdict/pull/2#pullrequestreview-804101121>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWGLYHNWEY2FGT6PZ6GYAETULQEABANCNFSM5HZWQWLQ>
> .
>

shinchanat committed:
Update change_value.py

shinchanat committed:
Update change_value.py

Fixed a bug in change value function

shinchanat committed:
Update README.md

shinchanat committed:
Delete change_value.py

shinchanat committed:
Add files via upload

This function is entirely different from the previous change_value fuction and more usefull.
No need to pass the keys in list,just pass the single relevant  key to change the relevant value.

shinchanat commented:
I changed the entire function , now no need to pass all the keys in list
Just pass a single relevant  key to change the value.

Example:

 Dic = {
              "cars":{
                     "bmw":1,
                      "tesla":2
                }
            }

print(nestdict.change_value(Dic,"tesla",10))



Output:

{"cars":{"bmw":1,"tesla":10}}


It returns original obj if element not found.

kindly go through it and let me know if any error or bug spotted.


On Fri, Nov 12, 2021, 11:36 AM Harish 2003 ***@***.***> wrote:

> Handled key error,  it returns None if element not found in obj.
>
> On Fri, Nov 12, 2021, 12:03 AM Abhishek Saini ***@***.***>
> wrote:
>
>> ***@***.**** requested changes on this pull request.
>>
>> Please make the changes to handle keyerror on line 51 of change value
>> function
>> ------------------------------
>>
>> In nestdict/change_value.py
>> <https://github.com/abhisaini880/nestdict/pull/2#discussion_r747728433>:
>>
>> > +       original : {"cars_owned":{"sedan":1,"suv":3},}
>> +
>> +       {"cars_owned":{"sedan":1,"suv":100},}
>> +
>> +
>> +
>> +      for more understanding read README.md
>> +      """
>> +
>> +    changed_obj = obj
>> +
>> +    keys = list(*args)
>> +
>> +    if len(keys)>1:
>> +        for element in keys:
>> +            obj = obj[element]
>>
>> return None or original obj if element not found in obj.
>> current code will throw key error.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/abhisaini880/nestdict/pull/2#pullrequestreview-804101121>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AWGLYHNWEY2FGT6PZ6GYAETULQEABANCNFSM5HZWQWLQ>
>> .
>>
>

shinchanat committed:
Update README.md

Included change_key docs

shinchanat committed:
Update README.md

shinchanat committed:
Update README.md

shinchanat committed:
Update README.md
